libobs/obs-module.c

```diff
@@ -177,6 +177,86 @@ void obs_log_loaded_modules(void)
 
 	for (obs_module_t *mod = obs->first_module; !!mod; mod = mod->next)
 		blog(LOG_INFO, "    %s", mod->file);
+#ifndef NDEBUG
+    // 1. enum all obs_source: input, filter, transition, scene
+    size_t idx = 0;
+    const char *id = NULL;
+    const char *unversioned_id = NULL;
+    
+    blog(LOG_INFO, "  1. enum all obs_source: input, filter, transition, scene");
+    
+    idx = 0;
+    blog(LOG_INFO, "    1.0. obs_enum_source_types()");
+    while(obs_enum_source_types(idx, &id))
+    {
+        blog(LOG_INFO, "      %02zd : id=%s, name=%s", idx, id, obs_source_get_display_name(id));
```

Comment:
You can probably omit the following line and just use `idx++` here, as it would be accessed and _then_ incremented.